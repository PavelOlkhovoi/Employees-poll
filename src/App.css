.App {
  text-align: center;
}

.logo {
  color: blue;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: 600;
}

.container {
  padding-top: 26px;
  max-width: 500px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

/*               MENU                */

header {
  padding: 30px 12px;
  background-color: white;
}

nav {
  display: flex;
  justify-content: space-between;
  max-width: 1000px;
  margin: 0 auto;
}

.menu-col {
  flex-basis: 30%;
}

.menu-col__centre {
  flex-grow: 1;
  margin: 0 auto;
  text-align: center;
  flex-basis: 40%;
}

.menu-col__left {
  margin-right: auto;
  text-align: left;
}

.menu-col__right {
  margin-right: auto;
  text-align: right;
}

.nav-link {
  padding: 6px;
}

.nav-link-active {
  color: rgb(54, 232, 10);
}


/*            POLL PREVEIW           */

.polls_cover {
  display: flex;
  align-items: center;
  justify-content: center;
}

.poll-previes {
  max-width: 400px;
  margin: 18px auto;
  background: white;
  border-radius: 8px;
  padding: 40px 60px;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  position: relative;
}

.poll-prev-author {
  margin: 20px 0 0 0;
  font-weight: 600;
}

.poll-prev-author-name {
  text-transform: capitalize;
  color: blue;
}

.poll-prev-votes {
  position: absolute;
  max-width: 200px;
  padding: 6px 18px;
  text-align: center;
  top: -20px;
  right: -30px;
  border: 2px solid rgb(58, 203, 58);
  background: transparent;
  border-radius: 300px;
  font-size: 12px;
  color: gray;
}





/*               POLL             */


.answer {
  background: white;
  display: flex;
  align-items: center;
  padding: 12px 20px;
  align-items: left;
  margin: 16px 0;
  border-radius: 8px;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
}

.poll-label {
  margin-right: 8px;
}

.poll-label::first-letter {
  text-transform: capitalize;
}

.info {
  margin-top: 20px;
  color: brown;
}

.info-autor {
  text-transform: capitalize;
}

.info-autor-name {
  color: blue;
  font-weight: 600;
}


/*             LEADERBOARD             */

table {
  text-align: left;
  width: 100%;
  border-collapse: collapse;
  background: white;
  border-radius: 8px;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
}

tr {
  padding: 12px;
}

td {
  padding: 12px;
}

th {
  padding: 12px;
}

tr:nth-child(1)  {
  background: blue;
  color: wheat;
  border-top-left-radius: 80px;
}

tr:first-child td:first-child { border-top-left-radius: 8px; }
tr:first-child td:last-child { border-top-right-radius: 8px; }


th:first-child {
  padding: 12px;
  border-top-left-radius: 8px;
}


th:last-child {
  padding: 12px;
  border-top-right-radius: 8px;
}


tr:nth-child(even)  {
  background-color: #f2f2f2;
}

/* tr {
  background: blue;
  color: white;
} */