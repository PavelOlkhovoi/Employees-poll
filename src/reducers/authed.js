import { SET_AUTHED_USER, CHANGE_AUTHED_USER, LOGOUT_USER } from "../actions/authed";

const authedReducer = (state = {status: null}, action) => {
    switch (action.type) {
        case SET_AUTHED_USER:
            return {...state, status: action.id}
        case CHANGE_AUTHED_USER: 
            return {...state, status: action.id}
        case LOGOUT_USER:
            return {...state, status: null}
        default:
            return state
    }
}

export default authedReducer;