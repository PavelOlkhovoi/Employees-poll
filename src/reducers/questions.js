import { RECEIVE_QUESTIONS, SAVE_ANSWER_POLL } from "../actions/questions";

const questions = (state={}, action) => {
    switch (action.type) {
        case RECEIVE_QUESTIONS:
            return {...state, ...action.questions}
        case SAVE_ANSWER_POLL:
            return {
                ...state,
                [action.pid]: {
                    ...state[action.pid],
                    [action.option]: {
                        ...state[action.pid][action.option],
                        votes: [...state[action.pid][action.option].votes, action.uid]
                    }
                }
            }
        default:
            return state
    }
}

export default questions;