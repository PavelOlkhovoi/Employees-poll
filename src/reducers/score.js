// import { CALCULATE_SCORE } from "../actions/score";

// // const scoreReducer = (state = {}, action) => {
// //     switch (action.type) {
// //         case CALCULATE_SCORE:
// //             return {
// //                 ...state,
// //                 ...action.score
// //             }

// //         default: state
// //         return state
// //     }
// // }

// // export default scoreReducer