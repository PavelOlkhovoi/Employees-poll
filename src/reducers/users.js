import { RECEIVES_USER, SAVE_ANSWER_USER, ADD_USER_POLL } from "../actions/users";

const usersReducer = (state = {}, action) => {
    switch (action.type) {
        case RECEIVES_USER:
            return {...state, ...action.users}
        case SAVE_ANSWER_USER:
            return {
                ...state,
                //id users
                [action.uid]: {
                    ...state[action.uid],
                    answers: {
                        //id question
                        ...state[action.uid].answers,
                        [action.pollId]: action.answer
                    }
                }
            }
        case ADD_USER_POLL:
            return {
                ...state,
                [action.uid]: {
                    ...state[action.uid],
                    questions: [
                        ...state[action.uid].questions,
                        action.pid
                    ]
                }
            }
        default:
            return state    
    }
}

export default usersReducer;