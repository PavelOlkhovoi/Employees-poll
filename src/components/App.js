import '../App.css';
import { connect } from 'react-redux';
import { useEffect } from 'react';
import initialData from '../actions/shared';
import { Routes, Route, useNavigate, Link, Navigate} from 'react-router-dom'
import { Fragment } from 'react';
import NavMenu from './NavMenu';
import Authorization from './Authorization';
import Home from './Home';
import Poll from './polls/Poll';
import CreatePoll from './polls/CreatePoll';
import LoadingBar from 'react-redux-loading-bar'


function App({dispatch, users, questions, authed, loading}) {
  const navigate = useNavigate()
  useEffect(()=> {
    dispatch(initialData())
  }, [])
  return (
    <Fragment>
      <div className="App">
        <NavMenu />
        <LoadingBar />
      </div>
      {
        authed.status === null ? 
        (
          // <div className='notice'>
          // If you want to use this app go to do
          // <Link to="/auth">Auth</Link>
          // </div>
          <Navigate to="/auth" />
        ) :
        "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN"
      }
      <Routes>
        <Route path='/' exact element={<Home />} />
        <Route path='/poll/:id' element={<Poll/>}/>
        <Route exact path='/auth' element={<Authorization />}/>
        <Route exact path='/create' element={<CreatePoll />}/>
        
      </Routes>
    </Fragment>
  );
}

const mapStateToProps = (state) => {
  return {
    users: state.users,
    questions: state.questions,
    authed: state.authed,
  }
}

export default connect(mapStateToProps)(App);
