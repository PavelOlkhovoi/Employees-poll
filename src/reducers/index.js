import usersReducer from "./users";
import questions from "./questions";
import authedReducer from "./authed";
import { combineReducers } from "redux";
import { loadingBarReducer  } from "react-redux-loading-bar";

export const reducers = combineReducers({
    users: usersReducer,
    questions,
    authed: authedReducer,
    loadingBar: loadingBarReducer,
})