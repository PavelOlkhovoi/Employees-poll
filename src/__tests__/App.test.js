import React from "react";
import { createStore } from "redux";
import { render, fireEvent, cleanup, screen } from "@testing-library/react";
import middleware from '../middleware';
import { Provider } from 'react-redux';
import { reducers } from '../reducers';
import { MemoryRouter } from 'react-router';
import { connect } from 'react-redux';
import App from "../components/App";


const store = createStore(reducers, middleware)

describe('Tessing of testing', ()=> {
   
    it('Test Redux', ()=> {
        const {getByText} = render(
            <Provider store={store}>
                <MemoryRouter>
                    <App/>
                </MemoryRouter>
            </Provider>
        )
    
        expect(getByText(/Leaderboard/i)).toBeInTheDocument()
    
    })


    it('Test Redux Snapshot', ()=> {
          
        const component = render(
                <Provider store={store}>
                    <MemoryRouter>
                        <App />
                    </MemoryRouter>
                </Provider>
            )
              
        expect(component).toMatchSnapshot();
    })
})