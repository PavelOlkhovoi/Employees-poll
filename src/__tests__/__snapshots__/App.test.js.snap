// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tessing of testing Test Redux Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="container"
        >
          <nav>
            <a
              href="/"
            >
              Home
            </a>
            <span>
              Leaderboard
            </span>
            <span>
              UserName
            </span>
            <a
              class="btn"
              href="/auth"
            >
              Logout
            </a>
            <a
              class="btn"
              href="/create"
            >
              Create new poll
            </a>
            <hr />
          </nav>
        </header>
        <div
          style="direction: ltr;"
        >
          <div
            class=""
            style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
          />
          <div
            style="display: table; clear: both;"
          />
        </div>
      </div>
      <div
        class="notice"
      >
        If you want to use this app go to do
        <a
          href="/auth"
        >
          Auth
        </a>
      </div>
      <div
        class="container"
      >
        <h1>
          Hi, users
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="container"
      >
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <span>
            Leaderboard
          </span>
          <span>
            UserName
          </span>
          <a
            class="btn"
            href="/auth"
          >
            Logout
          </a>
          <a
            class="btn"
            href="/create"
          >
            Create new poll
          </a>
          <hr />
        </nav>
      </header>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
    </div>
    <div
      class="notice"
    >
      If you want to use this app go to do
      <a
        href="/auth"
      >
        Auth
      </a>
    </div>
    <div
      class="container"
    >
      <h1>
        Hi, users
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
